src/clusterfuzz/_internal/base/errors.py

```diff
@@ -50,8 +50,8 @@ class Error(Exception):
 class InvalidTestcaseError(Error):
   """Error thrown when there is an attempt to access an invalid test case."""
 
-  def __init__(self):
-    super().__init__('Invalid test case.')
+  def __init__(self, testcase_id):
+    super().__init__(f'Invalid test case {testcase_id!r}.')
```

Comment:
what is `!r`?

Reply:
It formats with `repr()`, so that `"invalid-garbage"` would be rendered as `Invalid test case "invalid-garbage".`.

Reply:
Ah so you can omit the quotes?

Reply:
`repr(foo)` basically returns a string that evaluates to `foo` (at least for most basic types): https://docs.python.org/3/library/functions.html#repr. It's useful for debugging, since you can tell the difference between 12 and "12", b"12", Error("12"), etc.